pkg/apis/apps/v1alpha1/servicebinding_types.go

```diff
@@ -4,21 +4,21 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
 // ServiceBindingSpec defines the desired state of ServiceBinding
 // +k8s:openapi-gen=true
 type ServiceBindingSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
 	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
+	CSVName          string `json:"csvName"`
+	CSVNamespace     string `json:"csvNamespace"`
+	ApplicationLabel string `json:"ApplicationLabel"`
```

Comment:
Do we need multiple labels? I mean should we make it `[]string` as type here?

Reply:
shall we make ApplicationLabel as applicationLabel, just to maintain the consistency like csvNamespace and csvName?

Reply:
@Avni-Sharma Yes, I will change that. Thanks!

Reply:
We just want to bind to a backing service right. (No message queue or cache, I suppose). If it's only a db then I guess one label should suffice.

Reply:
What if there are multiple applications? Create separate CRs?

Reply:
After some thinking, I realize it's better to create separate CRs for different applications. Otherwise deployments will be re-deployed across for all labels.

Reply:
Multiple labels can be used to qualify service binding request to narrow down the scope and target a specific application development. This is the reason why there are multiple labels for label selectors in Kubernetes. I think we can follow the same approach. For example:
```
connects-to: postgres
environment: production
```
Another could be like this:
```
connects-to: postgres
environment: stage
```

Reply:
https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors

Reply:
How about something like this:
```
applicationSelector:
  matchLabels:
    connects-to: postgres
    environment: stage
```

Reply:
How about introducing resource "kind" also in the application selector?

For Development kind:
```
applicationSelector:
  matchLabels:
    connects-to: postgres
    environment: stage
  kind: Deployment
```

For StatefulSet kind:
```
applicationSelector:
  matchLabels:
    connects-to: postgres
    environment: stage
  kind: StatefulSet
```

Reply:
Instead of `kind`, probably we could qualify it further with a prefix like this: `resourceKind`. Or is it too verbose? ;-)

Reply:
I think so that resourceKind would cause less confusion because kind would be already there. Wdyt @baijum ?

Reply:
Done. 71aec326ad6ab64d910eec26628b29e7e4826a4b